src/makielayout/blocks/polaraxis.jl

```diff
@@ -0,0 +1,537 @@
+    @assert N ≤ 3
+    return to_ndim(
+        typeof(point),
+        Makie.project(
+            # obtain the camera of the Scene which will project to its screenspace
+            camera(scene),
+            # go from dataspace (transformation applied to inputspace) to pixelspace
+            :data, :pixel,
+            # apply the transform to go from inputspace to dataspace
+            Makie.apply_transform(
+                scene.transformation.transform_func[],
+                point
+            )
+        ),
+        0.0
+    )
+end
+
+function project_to_pixelspace(scene, points::AbstractVector{Point{N, T}}) where {N, T}
+    to_ndim.(
+        Ref(eltype(points)),
+        Makie.project.(
+            # obtain the camera of the Scene which will project to its screenspace
+            Ref(Makie.camera(scene)),
+            # go from dataspace (transformation applied to inputspace) to pixelspace
+            Ref(:data), Ref(:pixel),
+            # apply the transform to go from inputspace to dataspace
+            Makie.apply_transform(
+                scene.transformation.transform_func[],
+                points
+            )
+        ),
+        Ref(0.0)
+    )
+end
+
+# A function which redoes text layouting, to provide a bbox for arbitrary text.
+
+function text_bbox(textstring::AbstractString, textsize::Union{AbstractVector, Number}, font, align, rotation, justification, lineheight, word_wrap_width = -1)
+    glyph_collection = Makie.layout_text(
+            textstring, textsize,
+            font, align, rotation, justification, lineheight,
+            RGBAf(0,0,0,0), RGBAf(0,0,0,0), 0f0, word_wrap_width
+        )
+
+    return Rect2f(Makie.boundingbox(glyph_collection, Point3f(0), Makie.to_rotation(rotation)))
+end
+
+function text_bbox(plot::Text)
+    return text_bbox(
+        plot.text[],
+        plot.textsize[],
+        plot.font[],
+        plot.align[],
+        plot.rotation[],
+        plot.justification[],
+        plot.lineheight[],
+        RGBAf(0,0,0,0), RGBAf(0,0,0,0), 0f0,
+        plot.word_wrap_width[]
+    )
+end
+
+# Makie.can_be_current_axis(ax::PolarAxis) = true
+
+function Makie.initialize_block!(po::PolarAxis)
+    cb = po.layoutobservables.computedbbox
+
+    square = lift(cb) do cb
+        # find the widths of the computed bbox
+        ws = widths(cb)
+        # get the minimum width
+        min_w = minimum(ws)
+        # the scene must be a square, so the width must be the same
+        new_ws = Vec2f(min_w, min_w)
+        # center the scene
+        diff = new_ws - ws
+        new_o = cb.origin - 0.5diff
+        new_o =
+        Rect(round.(Int, new_o), round.(Int, new_ws))
+    end
+
+    scene = Scene(po.blockscene, square, camera = cam2d!, backgroundcolor = :transparent)
+
+    translate!(scene, 0, 0, -100)
+
+    Makie.Observables.connect!(
+        scene.transformation.transform_func,
+        @lift(PolarAxisTransformation($(po.θ_0), $(po.direction)))
+    )
```

Comment:
Should just be `connect!(...)` or `Observables.connect!(...)` if necessary. There are a couple of these around.